src/masonite/tests/TestCase.py

```diff
@@ -152,7 +155,15 @@ def fetch(self, path, data=None, method=None):
             return self.application.make("tests.response").build(
                 self.application, request, response, route
             )
-        raise Exception(f"No route found for url {path}")
+
+        exception = RouteNotFoundException(f"No route found for url {path}")
+        if self._exception_handling:
+            response = self.application.make("exception_handler").handle(exception)
```

Comment:
This way when exception handling is enabled but route not found during tests we can use assertions !